addons/website_sale/static/src/js/website_sale_options.js

```diff
@@ -188,6 +189,45 @@ sAnimations.registry.WebsiteSaleOptions = sAnimations.Class.extend(ProductConfig
     },
 });
 
-return sAnimations.registry.WebsiteSaleOptions;
+sAnimations.registry.productsSearchBar = sAnimations.Class.extend({
+    selector: 'section.website_sale_products_searchbar',
+    read_events: {
+        'keydown .o_search_products': '_onSearch',
+        'click .js_order_by': '_onOrderByClick',
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _onSearch: function (ev) {
+        wUtils.autocompleteWithProducts(this, this.$(ev.currentTarget), {
```

Comment:
Your code here initializes the autocomplete feature each time the user types a letter. This should be only at the "animation" start.
In edit mode, if the orderBy/displayPrice/etc option changes, you have to restart the animation on the element. To do that, you have to trigger an event. Check: `trigger_up('animation_start_demand'` in the code for example.